### ISSUE TITLE
`Mat.GetData(jagged: true)` returns multidimensional array, not jagged array

### ISSUE THREAD

**Issue by jskang-cle** [emgucv/emgucv#977]
```cs
    // Summary:
    //     Get a copy of the data values as an array
    //
    // Parameters:
    //   jagged:
    //     If true, a jagged array will returned. Otherwise it will return a regular array.
    //
    //
    // Returns:
    //     a copy of the data values as an array
    public Array GetData(bool jagged = true) { /* ... */ }
```

The `Mat.GetData` method claims to return a jagged array (e.g. `double[][]`) when the jagged parameter is set to true. However, the actual implementation of the method returns a multidimensional array (e.g. `double[,]`) instead of the expected jagged array.

**Comment by emgucv** [comment]
You're right — the documentation (and the parameter name itself) was incorrect: `GetData(true)` has always returned a multi-dimensional array (`byte[,]`, `byte[,,]`), never a jagged array, and that behavior is intentional (shape-preserving, matching the Mat's dimensions). For the 5.0 release the parameter has been renamed to `multiDimensional` with corrected documentation (1a9e143b26d6d9790177215a0f3fd2269fe38f66), so the API now says what it does: `GetData(multiDimensional: true)` → shaped multi-dimensional array, `false` → flat one-dimensional array. Behavior is unchanged. Closing as fixed for 5.0 — thanks for the report.
